drivers/include/mma7660.h

```diff
@@ -0,0 +1,200 @@
+ * for information about the values of the parameters
+ *
+ * @return                   0 on success
+ * @return                  -1 on error
+ */
+int mma7660_set_mode(mma7660_t *dev, uint8_t active,
+    uint8_t autowake, uint8_t autosleep, uint8_t prescale);
+
+/**
+ * @brief   Read the tilt register
+ *
+ * @param[in] dev          device descriptor
+ * @param[out] res         tilt register contents
+ *
+ * @return                   0 on success
+ * @return                  -1 on error
+ */
+int mma7660_read_tilt(mma7660_t *dev, uint8_t *res);
+
+/**
+ * @brief   Write the sleep count register
+ *
+ * @param[in] dev          device descriptor
+ * @param[in] sleep        sleep count
+ *
+ * @return                   0 on success
+ * @return                  -1 on error
+ */
+int mma7660_write_sleep_count(mma7660_t *dev, uint8_t sleep);
+
+/**
+ * @brief   Configure the interrupt sources
+ *
+ * @param[in] dev             device descriptor
+ * @param[in] isource_flags   interrupt source flags
+ *
+ * @return                   0 on success
+ * @return                  -1 on error
+ */
+int mma7660_config_interrupts(mma7660_t *dev, uint8_t isource_flags);
+
+/**
+ * @brief   Configure the sample rate
+ *
+ * @param[in] dev             device descriptor
+ *
+ * See page 18 of http://www.nxp.com/files/sensors/doc/data_sheet/MMA7660FC.pdf
+ * for details about the parameters
+ *
+ * @return                   0 on success
+ * @return                  -1 on error
+ */
+int mma7660_config_samplerate(mma7660_t *dev, uint8_t amsr, uint8_t awsr, uint8_t filt);
+
+/**
+ * @brief   Configure the tap detection
+ *
+ * @param[in] dev             device descriptor
+ *
+ * See page 21 of http://www.nxp.com/files/sensors/doc/data_sheet/MMA7660FC.pdf
+ * for details about the parameters
+ *
+ * @return                   0 on success
+ * @return                  -1 on error
+ */
+int mma7660_config_pdet(mma7660_t *dev, uint8_t pdth, uint8_t enabled_axes);
+
+/**
+ * @brief   Configure the tap detection debounce count
+ *
+ * @param[in] dev             device descriptor
+ *
+ * See page 21 of http://www.nxp.com/files/sensors/doc/data_sheet/MMA7660FC.pdf
+ * for details about the debouncer
+ *
+ * @return                   0 on success
+ * @return                  -1 on error
+ */
+int mma7660_config_pd(mma7660_t *dev, uint8_t pd) ;
+
+/**
+ * @brief   Configure the tap detection debounce count
+ *
+ * @param[in] dev             device descriptor
+ * @param[out] x              the X axis value
+ * @param[out] y              the Y axis value
+ * @param[out] z              the Z axis value
+ *
+ * See page 28 of http://www.nxp.com/files/sensors/doc/data_sheet/MMA7660FC.pdf
+ * for conversion of acceleration counts to angles or G forces.
+ *
+ * @return                   0 on success
+ * @return                  -1 on error
+ */
+int mma7660_read(mma7660_t *dev, int8_t *x, int8_t *y, int8_t *z);
```

Comment:
is the accelleromter only sampling with 8-bit?

In any case, I would strongly prefer to output the accel values in their physical representation in mili G [mg]:
```c
void mma7660_read(mma7660_t *dev, mma7660_data_t *data);

// with
typedef struct {
    int16_t x;
    int16_t y;
    int16_t z;
} mma7660_data_t;
```